File: m343l/hw1/p1.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main()
{
    int x,y;
    ifstream fin;
    fin.open("input.txt");
    ofstream fout;
    fout.open("output.txt");

    fin >> x >> y;
    if((x < 0) || (y < 0))
    {
        return -1;
    }
    else if((x == 0) || (y == 0))
    {
        return 0;
    }

    int r = x % y;
    while(r > 0)
    {
        x = y;
        y = r;  
        r = x % y;
    }
    fout << y << endl;
    return 0;
}

File: m343l/hw5/p1.py
#!/usr/bin/env

fin = open("input.txt", 'r');
fout = open("output.txt", 'w');
p = int(fin.next())
m = int(fin.next())
q1 = int(fin.next())
q2 = int(fin.next())

a = (q1 + q2) % m
b = (q1 * q2) % m
a = a % m
b = b % m

print >> fout, a,b

fin.close()
fout.close()

